packages/nimble-components/src/select/index.ts

```diff
@@ -199,6 +199,23 @@ export class Select
         return !(this.multiple || typeof this.size === 'number');
     }
 
+    /** @internal */
+    public labelSlot!: HTMLSlotElement;
+
+    /** @internal */
+    @volatile
+    public get labelContent(): string {
+        if (!this.$fastController.isConnected) {
+            return '';
+        }
+
+        const nodes = this.labelSlot.assignedNodes();
+        return nodes
+            .filter(node => node.textContent?.trim() !== '')
+            .map(node => node.textContent?.trim())
+            .join(' ');
```

Comment:
We have this logic duplicated slightly differently a couple times. Made a tech debt to clean it up. Could help to add this example to the issue so it's found easier after merge: https://github.com/ni/nimble/issues/2322

Reply:
Done